src/main/java/at/hannibal2/skyhanni/features/inventory/caketracker/CakeTracker.kt

```diff
@@ -0,0 +1,372 @@
+        return true
+    }
+
+    @SubscribeEvent
+    fun onBackgroundDraw(event: GuiRenderEvent.ChestGuiOverlayRenderEvent) {
+        if (!isEnabled()) return
+        if (inCakeBag || (inAuctionHouse && (unobtainedCakesDisplayed || searchingForCakes))) {
+            tracker.renderDisplay(config.cakeTrackerPosition, displayModeToggleable = false)
+        }
+    }
+
+    @SubscribeEvent
+    fun onBackgroundDraw(event: GuiContainerEvent.BackgroundDrawnEvent) {
+        if (!isEnabled()) return
+        if (inAuctionHouse) {
+            unobtainedCakesDisplayed = getCakeTrackerData()?.let { data ->
+                InventoryUtils.getItemsInOpenChest().filter {
+                    cakeNamePattern.matches(it.stack.displayName)
+                }.onEach { cakeItem ->
+                    cakeNamePattern.matchMatcher(cakeItem.stack.displayName) {
+                        group("year").toInt().takeIf { it !in data.cakesOwned }?.let {
+                            cakeItem highlight config.auctionHighlightColor.toChromaColor()
+                        }
+                    }
+                }.isNotEmpty()
+            } ?: false
+        }
+        if (inCakeInventory) checkInventoryCakes()
+    }
+
+    @SubscribeEvent
+    fun onInventoryOpen(event: InventoryFullyOpenedEvent) {
+        if (!isEnabled()) return
+        knownCakesInCurrentInventory.clear()
+        val inventoryName = event.inventoryName
+        if (cakeContainerPattern.matches(inventoryName)) {
+            if (cakeBagPattern.matches(inventoryName)) inCakeBag = true
+            val items = event.inventoryItems.filter { cakeNamePattern.matches(it.value.displayName) }
+            knownCakesInCurrentInventory = items.values.mapNotNull { item ->
+                cakeNamePattern.matchMatcher(item.displayName) {
+                    group("year")?.toInt()
+                }
+            }.toMutableList()
+            for (item in items) {
+                cakeNamePattern.matchMatcher(item.value.displayName) {
+                    val cakeYearInt = group("year").toInt()
+                    addCake(cakeYearInt)
+                }
+            }
+            inCakeInventory = true
+            timeOpenedCakeInventory = System.currentTimeMillis()
+            tracker.firstUpdate()
+        }
+        if (auctionBrowserPattern.matches(inventoryName)) {
+            inAuctionHouse = true
+            searchingForCakes = auctionCakeSearchPattern.matches(inventoryName)
+        } else inAuctionHouse = false
+    }
+
+    @SubscribeEvent
+    fun onInventoryClose(event: InventoryCloseEvent) {
+        inCakeBag = false
+        inCakeInventory = false
+        knownCakesInCurrentInventory.clear()
+        inAuctionHouse = false
+        unobtainedCakesDisplayed = false
+        searchingForCakes = false
+    }
+
+    @SubscribeEvent
+    fun onSecondPassed(event: SecondPassedEvent) {
+        if (!isEnabled()) return
+        if (currentYear == SkyBlockTime.now().year) return
```

Comment:
make a variable with SkyBlockTime.now() so you dont have to call it so many times in the same function